File: LeetCode/python3/215.py
class Solution:
    def findKthLargest(self, nums: List[int], k: int) -> int:
        min_heap = []
        for m in nums:
            if len(min_heap) < k or min_heap[0] < m:
                heapq.heappush(min_heap, m)
            if len(min_heap) > k:
                heapq.heappop(min_heap)
        return min_heap[0]

File: LeetCode/cpp/215.cpp
class Solution {
public:
    int findKthLargest(vector<int> &nums, int k) {
        priority_queue<int, vector<int>, greater<>> min_heap;
        for (auto &m : nums) {
            if (min_heap.size() < k || min_heap.top() < m)
                min_heap.push(m);
            if (min_heap.size() > k)
                min_heap.pop();
        }
        return min_heap.top();
    }
};
